packages/nextclade/src/run/validate_ref_seq.rs

```diff
@@ -0,0 +1,66 @@
+use crate::alphabet::nuc::Nuc;
+use crate::analyze::letter_ranges::{find_letter_ranges, NucRange};
+use crate::make_error;
+use color_eyre::{Section, SectionExt};
+use eyre::{Report, WrapErr};
+use itertools::Itertools;
+
+pub fn validate_ref_seq(seq_name: &str, seq: &[Nuc]) -> Result<(), Report> {
+  validate_ref_seq_impl(seq)
+    .wrap_err_with(|| format!("When validating reference sequence '{seq_name}'"))
+    .with_section(|| REF_SEQ_ERROR.header("Context:"))
+}
+
+fn validate_ref_seq_impl(seq: &[Nuc]) -> Result<(), Report> {
+  seq_check_empty(seq)?;
+  seq_check_gaps(seq)?;
+  Ok(())
+}
+
+fn seq_check_empty(seq: &[Nuc]) -> Result<(), Report> {
+  if seq.is_empty() {
+    return make_error!("Sequence is empty");
+  }
+  Ok(())
+}
+
+fn seq_check_gaps(seq: &[Nuc]) -> Result<(), Report> {
+  let ref_gap_ranges: Vec<NucRange> = find_letter_ranges(seq, Nuc::Gap);
+  if !ref_gap_ranges.is_empty() {
+    let ranges_formatted = ref_gap_ranges.into_iter().map(|r| r.range.to_string()).join(",");
+    return make_error!("Sequence contains gaps in the following positions: {ranges_formatted}. However, reference sequence is expected to be a high-quality, complete, and unambiguous as much as possible. Gaps are not allowed.");
```

Comment:
Rewording suggestion:

```suggestion
    return make_error!("Sequence contains gaps in the following positions: {ranges_formatted}. Gaps are not allowed: reference sequence is expected to be high-quality, complete, and unambiguous.");
```